docs/source/killed.rst

```diff
@@ -0,0 +1,105 @@
+.. _killed:
+
+Why did my worker die?
+----------------------
+
+A Dask worker can cease functioning for a number of reasons. These fall into the
+following categories:
+
+- the worker chooses to exit
+- an unrecoverable exception happens within the worker
+- the worker process is shut down by some external action
+
+Each of these cases will be described in more detail below. The *symptoms* you will
+experience when these things happen range from simply work not getting done anymore,
+to various exceptions appearing when you interact with your local client, such as
+``KilledWorker``, ``TimeoutError`` and ``CommClosedError``.
+
+Note the special case of ``KilledWorker``: this means that a particular task was
+tried on a worker, and it died, and then the same task was sent to another worker,
+which also died. After a configurable number of deaths, Dask decides to blame the
+task itself, and returns this exception.
+
+In every case, the first place to look for further information is the logs of
+the given worker, which may well give a complete description of what happened. These
+logs are printed by the worker to it's "standard output", which may be the text
+console from which you launched the worker, or some logging system maintained by
+the cluster infrastructure.
+
+In all cases, the scheduler will notice that the worker has gone, either because
+of an explicit de-registration, or because the worker no longer produces heartbeats,
+and it should be possible to reroute tasks to other workers and have the system
+keep running.
+
+Worker chose to exit
+''''''''''''''''''''
+
+Workers may exit in normal functioning because they have been asked to, e.g.,
+they received a keyboard interrupt (^C), or the scheduler scaled down the cluster.
+In such cases, the work that was being done by the worker will be redirected to
+other workers, if there are any left.
+
+You should expect to see the following message at the end of the worker's log:
+
+.. pull-quote::
+
+   distributed.dask_worker - INFO - End worker
+
+In these cases, there is not normally anything which you need to do, since the
+behaviour is expected.
+
+Unrecoverable Exception
+'''''''''''''''''''''''
+
+The worker is a python process, and like any other code, an exception may occur
+which causes the process to exit. One typical example of this might be a
```

Comment:
Is version mismatch likely now that we have version checking enabled on startup?

Reply:
Yes I think so, in the case that they don't have the latest versions anyway (which is pretty common)